src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1673,20 +1673,33 @@ private BaseListSyntax ParseBaseList()
             {
                 // first type
                 TypeSyntax firstType = this.ParseType();
-                list.Add(_syntaxFactory.SimpleBaseType(firstType));
+
+                ArgumentListSyntax argumentList = null;
+                if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+                {
+                    argumentList = this.ParseParenthesizedArgumentList();
+
+                    if (classOrStructOrInterfaceKeyword.Kind != SyntaxKind.ClassKeyword || !haveParameters)
```

Comment:
>!haveParameters [](start = 91, length = 15)

The specification at [https://github.com/dotnet/csharplang/pull/3527](https://github.com/dotnet/csharplang/pull/3527) does not support an error in the case where there are no parameters.

Reply:
> The specification at [https://github.com/dotnet/csharplang/pull/3527](https://github.com/dotnet/csharplang/pull/3527) does not support an error in the case where there are no parameters.

As it stands right now, parameters are required for a record declaration and only a class record declaration is allowed to have arguments. Given the current implementation of type parsing, presence of parameters is an indicator that we are parsing a record. This is a error recovery situation, we could not even attempt to parse the arguments, but I chose to parse them and report an error instead. Reconciling parsing of other parts of record declaration with the updated spec is out of scope for this PR. When that work will happen, necessary adjustments are going to be made to this code, if necessary.  

---
In reply to: [433447966](https://github.com/dotnet/roslyn/pull/44733#discussion_r433447966) [](ancestors = 433447966)